File: app/src/main/java/com/tomclaw/appsend/util/SchedulersFactory.java
package com.tomclaw.appsend.util;

import io.reactivex.rxjava3.android.schedulers.AndroidSchedulers;
import io.reactivex.rxjava3.core.Scheduler;
import io.reactivex.rxjava3.schedulers.Schedulers;

/**
 * Created by solkin on 20/01/2018.
 */
public interface SchedulersFactory {

    Scheduler io();

    Scheduler mainThread();

    class SchedulersFactoryImpl implements SchedulersFactory {

        @Override
        public Scheduler io() {
            return Schedulers.io();
        }

        @Override
        public Scheduler mainThread() {
            return AndroidSchedulers.mainThread();
        }

    }

}

File: app/src/main/java/com/tomclaw/appsend/util/SchedulersFactory.kt
package com.tomclaw.appsend.util

import io.reactivex.rxjava3.android.schedulers.AndroidSchedulers
import io.reactivex.rxjava3.core.Scheduler
import io.reactivex.rxjava3.schedulers.Schedulers

interface SchedulersFactory {

    fun io(): Scheduler

    fun mainThread(): Scheduler

}

class SchedulersFactoryImpl : SchedulersFactory {

    override fun io(): Scheduler = Schedulers.io()

    override fun mainThread(): Scheduler = AndroidSchedulers.mainThread()

}
